torchrl/envs/common.py

```diff
@@ -1741,10 +1750,21 @@ def __init__(
         self._constructor_kwargs = kwargs
         self._check_kwargs(kwargs)
         self._env = self._build_env(**kwargs)  # writes the self._env attribute
+        if self.batch_size in (None, torch.Size([])):
+            self.__dict__["_batch_size"] = self._get_batch_size(self._env)
         self._make_specs(self._env)  # writes the self._env attribute
         self.is_closed = False
         self._init_env()  # runs all the steps to have a ready-to-use env
 
+    def _get_batch_size(self, env):
```

Comment:
Can you explain a bit why this is needed?

In vmas the batch size is modified in the wrapper

https://github.com/pytorch/rl/blob/43b943e9fc238fd2515d8d1caf190bc380b9c04d/torchrl/envs/libs/vmas.py#L147

I would rather do something like that that introducing a new method

Reply:
What is the case against a method?
I guess most wrappers will want a dedicated way of computing the batch size. We sort of advertise that wrappers should have a way of looking at the env and determine the batch size accordingly.

Plus you can always do it yourself within _build_env 

Happy to refactor that within gym if it's too specific

Reply:
I see.
My point was just that adding a method may clutter a bit.
We have already so many methods to implement when creating a wrapper (make, build, init, ...)

I am ok if you want to introduce it but then I would present it as the standard way of doing this and uniform all our vecotrized env wrappers to use this (at the price of another method that you should know about if you are wrapping your vector env).

Reply:
I can remove it np